### DISCUSSION TITLE
Argo-rollouts not changing the image name for deployment referenced in workloadRef section

### DISCUSSION THREAD

**Discussion by preethignanesh** [argoproj/argo-rollouts#2161] [General]
I am trying the Argo rollouts for blue/green strategy. We have defined the deployments within our component spec, so while creating the rollout yaml instead of defining the pod config, I am using
workloadRef: 
    apiVersion: apps/v1
    kind: Deployment
    name: rollout-ref-deployment
to point to the existing deployment.
With this approach, rollout is creating new set of replicas specified in the rollout yaml instead of monitoring the existing deployed pods. Can't the rollout monitor the existing deployments instead of creating different replica set which in turn spins up new pods ?
 
Also when I try to rollout using the command - kubectl argo rollouts set image, its not changing the image tag of the existing deployment and the rollout is not progressing.
Is it possible to get the rollout command work when we have the rollout yaml pointing to the existing deployment ?

**Comment by kostis-codefresh** [comment]
Related https://github.com/argoproj/argo-rollouts/issues/1537
